libsolidity/formal/SMTEncoder.cpp

```diff
@@ -304,10 +307,49 @@ void SMTEncoder::endVisit(Block const& _block)
 
 bool SMTEncoder::visit(InlineAssembly const& _inlineAsm)
 {
+	struct AssignedExternalsCollector: public yul::ASTWalker
+	{
+		using ExternalReferences = map<yul::Identifier const*, InlineAssemblyAnnotation::ExternalIdentifierInfo>;
+		AssignedExternalsCollector(InlineAssembly const& _inlineAsm): externalReferences(_inlineAsm.annotation().externalReferences)
+		{
+			this->operator()(_inlineAsm.operations());
+		}
+
+		ExternalReferences const& externalReferences;
+		set<VariableDeclaration const*> assignedVars;
+
+		using yul::ASTWalker::operator();
+		void operator()(yul::Assignment const& _assignment)
+		{
+			auto const& vars = _assignment.variableNames;
+			for (auto const& identifier: vars)
+				if (auto it = externalReferences.find(&identifier); it != externalReferences.end())
```

Comment:
Can you replace this by `valueOrNullptr` from `libsolutil/CommonData.h`?

Reply:
Replaced.